util/kubeconfig/kubeconfig.go

```diff
@@ -94,6 +97,11 @@ func New(clusterName, endpoint string, caCert *x509.Certificate, caKey *rsa.Priv
 func CreateSecret(ctx context.Context, c client.Client, cluster *clusterv1.Cluster) error {
 	clusterCA, err := secret.Get(c, cluster, secret.ClusterCA)
 	if err != nil {
+		if apierrors.IsNotFound(err) {
+			return errors.Wrapf(&capierrors.RequeueAfterError{RequeueAfter: 30 * time.Second},
```

Comment:
This shouldn't return a RequeueAfterError because these utilities are exposed to other users as well. Instead, we should return a custom error defined in this package that users can match on and then requeue

Reply:
RequeueAfter shouldn't be needed anyway, since we already have a watch defined on Machines that filters down to "Control Plane" Machines.

Reply:
I was thinking this case, 

controlplane machine is deployed
cluster is queued for reconciliation
bootstrap provider hasn't created ca-cert secret yet

thats why i used RequeueAfterError, so it'll wait 30s for the bootstrap controller to create ca-cert secret and queued again. 

Is this a valid case though? It shouldn't take long for bootstrap provider to create secrets

Reply:
Yeah that makes sense, @tahsinrahman. I was pointing out though that the caller has to return handle RequeueAfterError, given that this function might be used elsewhere that doesn't understand that particular error

Reply:
thanks, i got it :) i've updated the pr, does it look right?

Reply:
Added a few more comments, ptal